zone/bot_raid.cpp

```diff
@@ -0,0 +1,2636 @@
+			default:
+				if (p_item && p_item->GetItem()->IsType2HWeapon()) {
+					melee_distance = melee_distance_max * 0.70f;
+				}
+				else {
+					melee_distance = melee_distance_max * 0.50f;
+				}
+
+				break;
+			}
+		}
+		float melee_distance_min = melee_distance / 2.0f;
+
+		// Calculate caster distances
+		float caster_distance_max = 0.0f;
+		float caster_distance_min = 0.0f;
+		float caster_distance = 0.0f;
+		{
+			if (GetLevel() >= GetStopMeleeLevel() && GetClass() >= WARRIOR && GetClass() <= BERSERKER) {
+				caster_distance_max = MAX_CASTER_DISTANCE[(GetClass() - 1)];
+			}
+
+			if (caster_distance_max) {
+
+				caster_distance_min = melee_distance_max;
+				if (caster_distance_max <= caster_distance_min) {
+					caster_distance_max = caster_distance_min * 1.25f;
+				}
+
+				caster_distance = ((caster_distance_max + caster_distance_min) / 2);
+			}
+		}
+
+		bool atArcheryRange = IsArcheryRange(tar);
+
+		if (GetRangerAutoWeaponSelect()) {
+
+			bool changeWeapons = false;
+
+			if (atArcheryRange && !IsBotArcher()) {
+
+				SetBotArcherySetting(true);
+				changeWeapons = true;
+			}
+			else if (!atArcheryRange && IsBotArcher()) {
+
+				SetBotArcherySetting(false);
+				changeWeapons = true;
+			}
+
+			if (changeWeapons) {
+				ChangeBotArcherWeapons(IsBotArcher());
+			}
+		}
+
+		if (IsBotArcher() && atArcheryRange) {
+			atCombatRange = true;
+		}
+		else if (caster_distance_max && tar_distance <= caster_distance_max) {
+			atCombatRange = true;
+		}
+		else if (tar_distance <= melee_distance) {
+			atCombatRange = true;
+		}
+
+		//#pragma endregion
+
+		//#pragma region ENGAGED AT COMBAT RANGE
+
+				// We can fight
+		if (atCombatRange) {
+
+			//if (IsMoving() || GetCombatJitterFlag()) { // StopMoving() needs to be called so that the jitter timer can be reset
+			if (IsMoving()) {
+
+				// Since we're using a pseudo-shadowstep for jitter, disregard the combat jitter flag
+				//if (!GetCombatJitterFlag()) {
+				StopMoving(CalculateHeadingToTarget(tar->GetX(), tar->GetY()));
+				//}
+
+				return;
+			}
+
+			// Combat 'jitter' code
+			// Note: Combat Jitter is disabled until a working movement solution can be found
+			if (AI_movement_timer->Check() && (!spellend_timer.Enabled() || GetClass() == BARD)) {
+
+				if (!IsRooted()) {
+
+					if (HasTargetReflection()) {
+
+						if (!tar->IsFeared() && !tar->IsStunned()) {
+
+							if (GetClass() == ROGUE) {
+
+								if (m_evade_timer.Check(false)) { // Attempt to evade
+
+									int timer_duration = (HideReuseTime - GetSkillReuseTime(EQ::skills::SkillHide)) * 1000;
+									if (timer_duration < 0) {
+										timer_duration = 0;
+									}
+
+									m_evade_timer.Start(timer_duration);
+									if (zone->random.Int(0, 260) < (int)GetSkill(EQ::skills::SkillHide)) {
+										RogueEvade(tar);
+									}
+
+									return;
+								}
+							}
+
+							//if (tar->IsRooted()) { // Move caster/rogue back from rooted mob - out of combat range, if necessary
```

Comment:
Remove comments if they don’t need to be here

Reply:
Thanks Akkadius.  I copied the bot.cpp as the basis for the bot_raid.cpp and these comments were pre-existing.  It appears that there is still work to be done in some bot behaviour.  I have deleted from bot_raid though likely requires more discussion.  Interested in your thoughts.

Reply:
I cleaned up this during my refactoring.